gcovr/json_generator.py

```diff
@@ -31,21 +49,65 @@ def print_json_report(covdata, output_file, options):
 
     for no in sorted(covdata):
         gcovr_json_file = {}
-        gcovr_json_file['file'] = presentable_filename(covdata[no].filename, root_filter=options.root_filter)
+        gcovr_json_file['file'] = presentable_filename(covdata[no].filename,
+                                                       root_filter=options.root_filter)
         gcovr_json_file['lines'] = _json_from_lines(covdata[no].lines)
         gcovr_json_root['files'].append(gcovr_json_file)
 
-    write_json = json.dump
-    if options.prettyjson:
-        write_json = functools.partial(write_json, indent=PRETTY_JSON_INDENT, separators=(',', ': '), sort_keys=True)
-    else:
-        write_json = functools.partial(write_json, sort_keys=True)
+    _write_json_result(gcovr_json_root, output_file, options)
 
-    if output_file is not None:
-        with open(output_file, 'w') as output:
-            write_json(gcovr_json_root, output)
-    else:
-        write_json(gcovr_json_root, sys.stdout)
+
+#
+# Produce gcovr JSON summary report
+#
+def print_json_summary_report(covdata, output_file, options):
+
+    total_lines = 0
+    total_covered = 0
+
+    json_dict = {}
+    json_dict['current_working_directory'] = options.root
+    json_dict['gcovr/format_version'] = JSON_FORMAT_VERSION
```

Comment:
This types of keys gcovr/* like gcovr/format_version were added as a some extension keys which don't exist in intermediate JSON format from GCOV. I suppose your JSON summary has nothing common with the GCOV intermediate JSON, so you can change the gcovr/format_version key to just the format_version (or summary_format_version) = JSON_SUMMARY_FORMAT_VERSION.
I would prefer to separate the JSON_FORMAT_VERSION and the JSON_SUMMARY_FORMAT_VERSION to emphasise that they have nothing common.